CenterForOpenScience/osf.io
# [ENG-1708] Creating a limited use raw metrics endpoint

## Purpose

Creating a proxy endpoint for accessing the raw elasticsearch endpoint.

## Changes

Adding a view and serializer to proxy this endpoint.
Adding permissions for the raw endpoint.
Adding a waffle switch to activate/deactivate this endpoint for prod.

## QA Notes

What features or workflows might this change impact? Any impact workflow

Does this change require a data migration? Yes. To add the new switch

If so, what data will we migrate?

If not, QA should still test/check existing and new?

What is the level of risk? Low as long as it doesn't run on Prod or for arbitrary users

Any permissions code touched? Kinda?

Is this an additive or subtractive change, other? Additive

How will QA verify? (UI, API, admin app or AdminAdminApp?) Dev test

Verify user tag never runs on Prod

How will this impact performance? Won't unless someone runs really tough queries on elastic

## Documentation

N/A

## Side Effects

Shouldn't be

## Ticket

https://openscience.atlassian.net/browse/ENG-1708

## Notes

In order to use this endpoint, the `ENABLE_RAW_METRICS` switch must be present (there is no migration for this switch) and active, and the user must have the `raw_es6` user tag.
